### DISCUSSION TITLE
Query about key decorator @key

### DISCUSSION THREAD

**Discussion by xw695** [microsoft/typespec#4287] [Q&A] [answered]
Hi, 
I want to know the usage of key decorator(@key). 
I have tried to add @key to 'id' property in my model and compiled with 2 emitters below. But the result is the same with model without @key. So it seems @key doesn't take effect in my case, why?

@azure-tools/typespec-ts
@typespec/openapi3

`model MyCalculationOutput {
  @key
  id: string;
  processDate: plainDate;
  result: string;
}

@get
@route("/$calculation")
op calculation(): MyCalculationOutput[] ;`

Thanks

**Comment by xw695** [comment]
The background is: The 'calculation' method returns a list of records. I want this list of records could be presented as a dictionary/map of records with the 'id' of each record used as the dictionary/map key

**Reply by bterlson** [reply, marked as answer]
Key conceptually flags the fields which identify the resource, but doesn't imply any special behavior for that type within an array. You can maybe get what you want  by doing:

```
op calculation(): Record<MyCalculationOutput>
```

**Reply by xw695** [reply]
Thanks, I think you are right, it doesn't work as I expected. I will find a workaround for my request~
